### ISSUE TITLE
Enable/Disable Tina from inside <Tina> context 

### ISSUE THREAD

**Issue by jamespohalloran** [tinacms/tinacms#754]
## Summary

I would like to enable/disable the Tina "toggle" button from within a page. 

### Basic example

I have a NextJS site with a main layout using: 
```

export default withTina(Site, {
  cms: {
// ...
  },
  sidebar: {
    hidden: true,
  },
})
```

From within a view, I have a "preview" prop (based from getStaticProps). This preview prop isn't available in the main layout. I would like to be able to toggle Tina's edit-mode from within this view based on this `preview` property.

### Motivation

Showing/Hiding the sidebar toggle is a bit tricky when it's not based on build environment variables

**Comment by ncphillips** [comment]
The [isOpen](https://github.com/tinacms/tinacms/blob/master/packages/tinacms/src/tina-cms.ts#L101-L108) API could be a good model:

```ts
get hidden() {
  return this._hidden
}
set hidden(val: boolean) {
  this._hidden = nextValue
  this.notifySubscribers()
}
```

**Comment by kendallstrautman** [comment]
Something like this - https://github.com/tinacms/tinacms/pull/769/commits/bf2956c3be73f9aff9843dcaeb0ac500d9cd285e - via #769?

**Comment by jamespohalloran** [comment]
Looks good to me!

**Comment by ncphillips** [comment]
Implemented by #769
